src/sentry/similarity/features.py

```diff
@@ -198,10 +90,16 @@ def __get_scope(self, group):
     def __get_key(self, group):
         return '{}'.format(group.id)
 
+    def extract(self, event):
+        return {
+            label: feature.extract(event) for label, feature in self.features.items()
+        }
+
     def record(self, event):
         items = []
-        for label, feature in self.features.items():
-            for characteristics in feature.extract(event):
+        for label, characteristics_list in self.extract(event).items():
+            for characteristics in characteristics_list:
```

Comment:
Note: we talked about updating this so that each feature can only return one characteristic, otherwise [scaling to the total](https://github.com/getsentry/sentry/blob/837019d667af7f137f7641e841cf080ffe9a7ff8/src/sentry/scripts/similarity/index.lua#L478-L479) doesn't really make any sense.